### PR TITLE
Refactor MSDeployArtifactHandler, extract common codes

### PR DESCRIPTION
**Pull request by Flanker32**
1. Refactor MSDeployArtifactHandler, extract common codes to `FunctionArtifactHelper`
2. Import PowerMockito to do unit test on static methods

### CODE REVIEW

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/Constants.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.maven.function;
+
+public class Constants {
+	public static final String ZIP_EXT = ".zip";
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/Constants.java]: space

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/MSDeployArtifactHandlerImpl.java [hunk 2]
```diff
@@ -70,7 +61,7 @@ private MSDeployArtifactHandlerImpl(@Nonnull final Builder builder) {
     public void publish(final DeployTarget target) throws Exception {
         final File zipPackage = createZipPackage();
 
-        final CloudStorageAccount storageAccount = getCloudStorageAccount(target);
+        final CloudStorageAccount storageAccount = FunctionArtifactHelper.getCloudStorageAccount(target,log);
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/MSDeployArtifactHandlerImpl.java]: one space  before log

File: azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+public class FunctionArtifactHelperTest {
+
+    @Test
+    public void getCloudStorageAccount() throws Exception {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelperTest.java]: testGetCloudStorageAccount

File: azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        doReturn(mapSettings).when(deployTarget).getAppSettings();
+        doReturn(storageSetting).when(mapSettings).get(anyString());
+        doReturn(storageConnection).when(storageSetting).value();
+        final Log log = mock(Log.class);
+        final CloudStorageAccount storageAccount = FunctionArtifactHelper.getCloudStorageAccount(deployTarget,log);
+        assertNotNull(storageAccount);
+    }
+
+    @Test
+    public void getCloudStorageAccountWithException() {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelperTest.java]: test prefix.

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+public class FunctionArtifactHelper {
+
+    private static final String STAGE_DIR_NOT_FOUND = "Azure Functions stage directory not found. " +
+            "Please run 'mvn package azure-functions:package' first.";
+    private static final String REMOVE_LOCAL_SETTINGS = "Remove local.settings.json from ZIP package.";
+    private static final String UNSUPPORTED_DEPLOYMENT_TARGET = "Unsupported deployment target, only function is supported";
+    private static final String INTERNAL_STORAGE_NOT_FOUND = "Application setting 'AzureWebJobsStorage' not found.";
+    private static final String INTERNAL_STORAGE_CONNECTION_STRING = "Azure Function App's Storage Connection String: ";
+
+    public static File createZipPackage(final String stagingDirectoryPath, final Log log) throws Exception {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: if the log never changes, consider to save it as an static variable

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        final File stageDirectory = new File(stagingDirectoryPath);
+        final File zipPackage = new File(stagingDirectoryPath.concat(Constants.ZIP_EXT));
+
+        if (!stageDirectory.exists()) {
+            log.error(STAGE_DIR_NOT_FOUND);
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: what's the behavior when stageDirectory is actually a file not a directory?

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java (hunk 5 above)
Comment (andxu) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: or we can remove it from args and catch it outside by log the error thrown

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    private static final String STAGE_DIR_NOT_FOUND = "Azure Functions stage directory not found. " +
+            "Please run 'mvn package azure-functions:package' first.";
+    private static final String REMOVE_LOCAL_SETTINGS = "Remove local.settings.json from ZIP package.";
+    private static final String UNSUPPORTED_DEPLOYMENT_TARGET = "Unsupported deployment target, only function is supported";
+    private static final String INTERNAL_STORAGE_NOT_FOUND = "Application setting 'AzureWebJobsStorage' not found.";
+    private static final String INTERNAL_STORAGE_CONNECTION_STRING = "Azure Function App's Storage Connection String: ";
+
+    public static File createZipPackage(final String stagingDirectoryPath, final Log log) throws Exception {
+        final File stageDirectory = new File(stagingDirectoryPath);
+        final File zipPackage = new File(stagingDirectoryPath.concat(Constants.ZIP_EXT));
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: when zipPackage  exists, is it well handled?

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java (hunk 5 above)
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: bad name of createZipPackage, createZipFile

Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: stagingDirectoryPath => directory

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: Removed it

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: Changed to `createFunctionArtifact`

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: keep `stagingDirectoryPath` as this is a function utils used to package function staging folder

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java (hunk 7 above)
Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: `ZipUtil` will overwrite the previous one

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java (hunk 6 above)
Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: it shouldn't package a file here, added `isDirectory()` check

**Pull request merged**